util/topgen/merge.py

```diff
@@ -747,6 +747,52 @@ def amend_resets(top, name_to_block):
     top["resets"] = top_resets
 
 
+def create_alert_lpgs(top, name_to_block: Dict[str, IpBlock]):
+    '''Loop over modules and determine number of unique LPGs'''
+    num_lpg = 0
+    lpg_dict = {}
+    top['alert_lpgs'] = []
+    for module in top["module"]:
+        # the alert senders are attached to the primary clock of this block,
+        # so let's start by getting that primary clock.
+        block = name_to_block[module['type']]
+        block_clock = block.get_primary_clock()
+        primary_reset = module['reset_connections'][block_clock.reset]
+
+        # for the purposes of alert handler LPGs, we need to know:
+        #   1) the clock group of the primary clock
+        #   2) the primary reset name
+        #   3) the domain of the primary reset
+        clock_group = module['clock_group']
+        reset_name = primary_reset['name']
+        reset_domain = primary_reset['domain']
+
+        # using this info, we can create an LPG identifier
+        # and uniquify it via a dict.
+        lpg_name = '_'.join([clock_group, reset_name, reset_domain])
+        if lpg_name not in lpg_dict:
+            lpg_dict.update({lpg_name: num_lpg})
+            # since the alert handler can tolerate timing delays on LPG
+            # indication signals, we can just use the clock / reset signals
+            # of the first block that belongs to a new unique LPG.
+            clock = module['clock_connections'][block_clock.clock]
+            top['alert_lpgs'].append({
+                'name': lpg_name,
+                'clock_group': clock_group,
+                'clock_connection': clock,
+                'reset_connection': primary_reset
+            })
+            num_lpg += 1
```

Comment:
i wonder if it makes sense to extend the current clocks class with this information.. or maybe it might make more sense to create an alerts class that pulls in clock objects. Then you can pick out specific clock signals without needing to reconstruct these.

Reply:
ok let me see... I believe it does not make too much sense to add it to the clock / reset classes, since the LPGs are formed when combining specific resets with a clock group. so probably it makes more sense to create an alert class to hold this info.

this will probably need some more refactoring of the alerts into class objects (besides the LPG stuff here).
I'll give this a try to see how much would have to change.

Reply:
o yeah i think this makes more sense as an alert class that references clocks / resets objects also.
i don't think we should bother now though..that feels likes a subsequent PR cleanup if we have time.

Reply:
yeah let's get these connections in first.